network_wrangler/transit/projects/add_route.py

```diff
@@ -89,71 +89,66 @@ def _add_route_to_feed(
+            feed.shapes = pd.concat([feed.shapes, add_shape_df], ignore_index=True, sort=False)
+            
+            for j, headway in enumerate(trip["headway_secs"]):
+                trip_id = f"trip{j}_shp{shape_id}"
+                stop_dicts = _get_stops_from_routing(trip["routing"])
+                add_trips_df = _create_new_trips(trip, route["route_id"], trip_id, shape_id)
+                add_freqs_df = _create_new_frequencies(headway, trip_id)
+                add_stop_times_df = _create_new_stop_times(stop_dicts, trip_id)
+                add_stops_df = _create_new_stops(
+                    add_stop_times_df["stop_id"], feed.stops["stop_id"], road_net
+                )
+
+                if route['agency_raw_name']:
+                    add_trips_df['agency_raw_name'] = route['agency_raw_name']
+                    add_shape_df['agency_raw_name'] = route['agency_raw_name']
+                    add_freqs_df['agency_raw_name'] = route['agency_raw_name']
+                    add_stop_times_df['agency_raw_name'] = route['agency_raw_name']
+            
+                # Add new data to existing dataframes
+                feed.trips = pd.concat([feed.trips, add_trips_df], ignore_index=True, sort=False)
+                feed.frequencies = pd.concat(
+                    [feed.frequencies, add_freqs_df], ignore_index=True, sort=False
+                )
+                feed.stops = pd.concat([feed.stops, add_stops_df], ignore_index=True, sort=False)
+                feed.stop_times = pd.concat(
+                    [feed.stop_times, add_stop_times_df], ignore_index=True, sort=False
+                )
 
     return feed
 
 
 def _create_new_trips(
-    trips: list[dict], route_id: str, shapes_df: paDataFrame[WranglerShapesTable]
+    trip: dict, route_id: str, trip_id: str, shape_id: str
 ) -> paDataFrame[TripsTable]:
     """Create new trips for a route.
 
     Args:
-        trips: List of trip dictionaries with trip_id, shape_id, and other trip information.
+        trips: Trip dictionaries with trip_id, shape_id, and other trip information.
         route_id: Route ID for the trips.
-        shapes_df: Shapes dataframe to get shape_id from.
+        trip_id: Trip ID for the trips.
+        shape_id: Shape ID for the trips.
     """
-    FILTER_OUT = ["routing", "headway_secs"]
-    add_trips_df = pd.DataFrame(
-        [{k: v for k, v in r.items() if k not in FILTER_OUT} for r in trips]
-    )
-    add_trips_df["route_id"] = route_id
-    if "shape_id" not in add_trips_df.columns:
-        add_trips_df["shape_id"] = None
-
-    if "trip_id" not in add_trips_df.columns:
-        add_trips_df["trip_id"] = None
-    add_trips_df["shape_id"] = fill_str_ids(add_trips_df["shape_id"], shapes_df["shape_id"])
-    add_trips_df["trip_id"] = add_trips_df["trip_id"].fillna(
-        add_trips_df["shape_id"].apply(lambda x: f"tr_shp{x}")
-    )
+    add_trips_df = pd.DataFrame([{
```

Comment:
@yueshuaing This is prescriptive such that only the specific properties you added will get added. the code I had on the left is generalized such that any value you added could be transferred which is consistent with GTFS model and our models as well.  Is there something that didn't work in the other method?

Reply:
The previous code didn't add all trips as trips are added before frequencies. For example, we need to add four trips: direction 0 AM and MD, direction 1 AM and MD. The previous code only adds two trips of the two direction to the trips table. I changed the sequence, for each trip, check the headway (which also has the time of day info), then add new trip for each time period. I was trying to use the previous function, but since I changed the sequence of adding trips, some of the info like the shape_id, trip_id, has already been defined in the previous steps, so I simplified this function `_create_new_trips`

Reply:
Technically in GTFS there doesn't need to be a single trip/frequency.  Just a single trip per shape.  Do/should we require a single trip per frequency in our data model? If so, we need to enforce that in the data model.  Can you please create an issue if so @yueshuaing ?